# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/06/13 10:13:46-04:00 bcollins@debian.org 
#   ieee1394: Fix possible NULL ptr dereference with calls to find_ctx()
#   
#   Submitted-by: Hiromasa YOSHIMOTO <yosimoto@limu.is.kyushu-u.ac.jp>
#   Signed-off-by: Ben Collins <bcollins@debian.org>
# 
# drivers/ieee1394/video1394.c
#   2004/06/13 10:12:39-04:00 bcollins@debian.org +4 -0
#   Fix possible NULL ptr dereference with calls to find_ctx()
# 
diff -Nru a/drivers/ieee1394/video1394.c b/drivers/ieee1394/video1394.c
--- a/drivers/ieee1394/video1394.c	2004-06-22 12:52:36 -07:00
+++ b/drivers/ieee1394/video1394.c	2004-06-22 12:52:36 -07:00
@@ -864,6 +864,7 @@
 			return -EFAULT;
 
 		d = find_ctx(&ctx->context_list, OHCI_ISO_RECEIVE, v.channel);
+		if (d == NULL) return -EFAULT;
 
 		if ((v.buffer<0) || (v.buffer>d->num_desc)) {
 			PRINT(KERN_ERR, ohci->host->id,
@@ -926,6 +927,7 @@
 			return -EFAULT;
 
 		d = find_ctx(&ctx->context_list, OHCI_ISO_RECEIVE, v.channel);
+		if (d == NULL) return -EFAULT;
 
 		if ((v.buffer<0) || (v.buffer>d->num_desc)) {
 			PRINT(KERN_ERR, ohci->host->id,
@@ -1009,6 +1011,7 @@
 			return -EFAULT;
 
 		d = find_ctx(&ctx->context_list, OHCI_ISO_TRANSMIT, v.channel);
+		if (d == NULL) return -EFAULT;
 
 		if ((v.buffer<0) || (v.buffer>d->num_desc)) {
 			PRINT(KERN_ERR, ohci->host->id,
@@ -1112,6 +1115,7 @@
 			return -EFAULT;
 
 		d = find_ctx(&ctx->context_list, OHCI_ISO_TRANSMIT, v.channel);
+		if (d == NULL) return -EFAULT;
 
 		if ((v.buffer<0) || (v.buffer>d->num_desc)) {
 			PRINT(KERN_ERR, ohci->host->id,
